client/webserver/site/src/js/app.js

```diff
@@ -496,6 +496,8 @@ export default class Application {
         break
       case 'walletstate':
       case 'walletconfig': {
+        // assets can be null if failed to connect to dex server.
+        if (!this.assets) return
```

Comment:
The `Assets` field of the `User` object should be populated regardless of any particular dex connection status. Are you sure this comment is true?

Reply:
If the server is off, it is not possible to know which assets they have, or is it?

Reply:
`this.assets` is just `this.user.assets`, which comes from the Go `core.User` type. The `Assets` field is a `map[uint32]*SupportedAsset`, which are assets that the client supports, not `dex.Asset`, which would be associated with a DEX. The big clue is that it is a field of `User`, not of `Exchange`. Look at `(*Core).assetMap` to see how the `User.Assets` map is made.